troposphere/static/js/components/images/detail/header/HeaderView.jsx

```diff
@@ -45,8 +53,49 @@ export default React.createClass({
     render: function() {
         let profile = stores.ProfileStore.get(),
             { image } = this.props,
-            buttonGroup;
+            expiredBadge,
+            buttonGroup,
+            launchButton;
+
+        // NOTE: this could be `context.hasExpiredPassword()` if
+        // the context module were to be used.
+        // * See also: `profile.id` usage below + FIXME
+        if (hasExpiredPassword(profile)) {
+            let style = {
+                position: "absolute",
+                color: "red",
+                background: "white",
+                borderRadius: "50%",
+                top: "-5px",
+                left: "-5px"
+            };
+            expiredBadge = (
+                <i className="glyphicon glyphicon-exclamation-sign" style={style} />
+            );
+            launchButton = (
+                    <button
+                        className="btn btn-primary launch-button"
+                        disabled={ image.isEndDated() }
```

Comment:
This is quite the corner case, but I added this line. So that if account is expired, and image is endated, the button will appear disabled.